File: test5.js
function findFee(t) {
    let hour = 0;
    let min = 0;
    let total = 0;
    if (t.split(" hours")[1] === undefined) {
        min = parseInt(t.split(" minutes")[0]);
    } else{
        hour = parseInt(t.split(" hours")[0]);
        min = parseInt(t.split(" hours")[1].split(" minutes")[0]);
    }
    total = 60*hour + min;
    if(total < 30){
         console.log("Free");
    } else if(total <= 180){
        console.log(Math.ceil(total/60)*10 + " bath");
    } else if(total <= 360){
        console.log(Math.ceil(total/60)*20 + " bath");
    } else if(total > 360){
        console.log(Math.ceil(total/1440)*200 + " bath");
    }
}
findFee("2 hours 10 minutes");

File: test5.go
package main

import (
	"fmt"
	"strings"
	"strconv"
	"math"
)

func findFee(t string) {
    hour := 0
    min := 0
    var total float64
    if strings.Index(t, "hours") == -1 {
        min, _ = strconv.Atoi(strings.Split(t, " minutes")[0])
    } else{
	hour, _ = strconv.Atoi(strings.Split(t, " hours")[0])
        min, _ = strconv.Atoi(strings.Split(strings.Split(t, " hours ")[1], " minutes")[0])
    }
    total = float64(60*hour + min)
    if total < 30 {
	fmt.Println("Free")
    } else if total <= 180 {
	tmp := strconv.Itoa(int(math.Ceil(total/60)*10))
	fmt.Println(tmp + " bath")
    } else if total <= 360 {
	tmp := strconv.Itoa(int(math.Ceil(total/60)*20))
	fmt.Println(tmp + " bath")
    } else if total > 360 {
	tmp := strconv.Itoa(int(math.Ceil(total/1440)*200))
	fmt.Println(tmp + " bath")
    }
}

func main() {
	findFee("4 hours 15 minutes")
}
